orangecontrib/text/vectorization/document_embedder.py

```diff
@@ -0,0 +1,193 @@
+    language : str
+        ISO 639-1 (two-letter) code of desired language.
+    aggregator : str
+        Aggregator which creates document embedding (single
+        vector) from word embeddings (multiple vectors).
+        Allowed values are Mean, Sum, Max, Min.
+    """
+
+    def __init__(self, language: str = 'en',
+                 aggregator: str = 'Mean') -> None:
+        lang_error = '{} is not a valid language. Allowed values: {}'
+        agg_error = '{} is not a valid aggregator. Allowed values: {}'
+        if language.lower() not in LANGUAGES:
+            raise ValueError(lang_error.format(language, ', '.join(LANGUAGES)))
+        self.language = language.lower()
+        if aggregator.lower() not in AGGREGATORS_L:
+            raise ValueError(agg_error.format(aggregator, ', '.join(AGGREGATORS)))
+        self.aggregator = aggregator.lower()
+
+        self._embedder = _ServerEmbedder(self.aggregator,
+                                         model_name='fasttext-'+self.language,
+                                         max_parallel_requests=100,
+                                         server_url='https://example.com',
+                                         # TODO set proper url
+                                         embedder_type='text')
+
+    def __call__(self, corpus: Corpus, copy: bool = True,
+                 processed_callback=None) -> Corpus:
+        """Adds matrix of document embeddings to a corpus.
+
+        Parameters
+        ----------
+        corpus : Corpus
+            Corpus on which transform is performed.
+        copy : bool
+            If set to True, a copy of corpus is made.
+
+        Returns
+        -------
+        Corpus
+            Corpus (original or a copy) with new features added.
+
+        Raises
+        ------
+        ValueError
+            If corpus is not instance of Corpus.
+        RuntimeError
+            If document in corpus is larger than
+            50 KB after compression.
+        """
+        if not isinstance(corpus, Corpus):
+            raise ValueError("Input should be instance of Corpus.")
+        corpus = corpus.copy() if copy else corpus
+        embs = self._embedder.embedd_data(
+            list(corpus.ngrams),
+            processed_callback=processed_callback)
+
+        dim = None
+        send_warning = False
+        for emb in embs:  # find embedding dimension
+            if emb is not None:
+                dim = len(emb)
+                break
+        # Check if some documents in corpus in weren't embedded
+        # for some reason. This is a very rare case.
+        inds = list()
+        for i, emb in enumerate(embs):
+            if emb is not None:
+                inds.append(i)
+            else:
+                embs[i] = np.zeros(dim) * np.nan
+                send_warning = True
+
+        variable_attrs = {
+            'hidden': True,
+            'skip-normalization': True,
+            'document-embedding-feature': True
```

Comment:
I would shorten 'document-embedding-feature'. It makes Data Table look excessively wide, IMO.

Reply:
I've changed it to 'embedding-feature'. The problematic cell now has the same width as the cell in Data Table obtained from Bag of Words widget.